pac4j-saml-db/src/main/java/org/pac4j/saml/dbclient/DbLoadedSamlClientConfiguration.java

```diff
@@ -0,0 +1,211 @@
+package org.pac4j.saml.dbclient;
+
+import org.opensaml.saml.common.xml.SAMLConstants;
+import org.pac4j.saml.client.SAML2ClientConfiguration;
+import org.pac4j.saml.storage.EmptyStorageFactory;
+import org.pac4j.saml.storage.SAMLMessageStorageFactory;
+
+
+/**
+ * An alternative to {@link SAML2ClientConfiguration}. Unlike {@link SAML2ClientConfiguration}, this class does not use paths to resources.
+ * Instead, it contains the data directly.
+ *
+ * Differences from {@link SAML2ClientConfiguration}:
+ * <ul>
+ * <li>The JKS keystore - contained as a byte array, not a path to a resource</li>
+ * <li>IdP metadata - contained as a string, not a path to a resource</li>
+ * <li>SP metadata - dropped completely</li>
+ * <li>Client name - added</li>
+ * </ul>
+ *
+ * TODO: It would be nice to have a common interface and maybe a common abstract parent too.
+ * 
+ * @author jkacer
+ */
+public class DbLoadedSamlClientConfiguration implements Cloneable {

```

Comment:
I'm working on it right now and I've just encountered a little issue. When loading the configuration in init(context), we must already know the name for which we are loading the data - the names are not known in advance, unlike in case of static configuration. This means all names must be somehow read in advance and a particular name must be given to the configuration object. We can read all names using the DAO, no problem here. Is this how you meant it?